ticket_migrate.py

```diff
@@ -0,0 +1,829 @@
+        """
+        for comment in (c for c in comments if c['t_id'] == self.t_id):
+            self._submitComment(self.commentFromTracData(comment))
+
+    def submitToProject(self):
+        """
+        Add an issue identified by the GitHub global `id`
+        to the configured column ID of a project.
+
+        API docs (very bad ones):
+        https://docs.github.com/en/rest/reference/projects#create-a-project-card
+        """
+        column_id = config.COLUMN_ID
+        url = f'https://api.github.com/projects/columns/{column_id}/cards'
+        data = {
+            'content_id': self.github_id,
+            'content_type': 'Issue'
+            }
+
+        if DRY_RUN:
+            print(f"Would post to {url}:")
+            pprint.pprint(data)
+            return
+
+        response = requests.post(
+            url=url,
+            headers={'accept': 'application/vnd.github.v3+json'},
+            json=data,
+            auth=(config.OAUTH_USER, config.OAUTH_TOKEN)
+            )
+
+        if response.status_code != 201:
+            print('Error: Issue not added to project column!')
+            print(response)
+            pprint.pprint(response.json())
+            import pdb
+            pdb.set_trace()
+
+    @classmethod
+    def fromTracData(
+            cls,
+            component,
+            owner,
+            summary,
+            description,
+            priority,
+            keywords,
+            **kwargs):
+        """
+        Create a GitHubRequest from Trac ticket data fields.
+        """
+        return cls(
+            owner=config.OWNER,
+            repo=get_repo(component),
+            trac_id=kwargs['t_id'],
+            title=summary,
+            body=get_body(description, data=kwargs),
+            closed=kwargs['status'] == 'closed',
+            milestone=None,
+            labels=get_labels(
+                component,
+                priority,
+                keywords,
+                kwargs['status'],
+                kwargs['resolution']
+                ),
+            assignees=get_assignees(owner),
+            )
+
+    @classmethod
+    def fromTracDataMultiple(cls, trac_data):
+        """
+        Generate GitHubRequests from an iterable of dicts of Trac tickets.
+        """
+        for ticket in trac_data:
+            yield cls.fromTracData(**ticket)
+
+    @staticmethod
+    def commentFromTracData(trac_data):
+        """
+        Convert Trac comment data to GitHub comment body as JSON.
+        """
+        author, _ = config.USER_MAPPING.get(
+            trac_data['author'],
+            (trac_data['author'], 'ignored-email-field')
+            )
+
+        body = (
+            f"Comment by {author} at {showtime(trac_data['c_time'])}.\n"
+            f"\n"
+            f"{parse_body(trac_data['newvalue'])}"
+            )
+
+        return {'body': body}
+
+    def _submitComment(self, comment_data):
```

Comment:
just a question... a lot of this code should not be reused in other places.

why bother with private method names ?

and `def submit_comment` or `def subbmitComment` ? )

Reply:
I don't like methods too big. It is easier to think about them if they do less.

In `submitMyComments`, the public high-level method, it iterates through tickets and calls the low-level method.
In `_submitComment`, the private low-level method, it submits a POST request if appropriate, and records that it happened.

> why bother with private method names ?

Would you just join the code into one big method? Or just make everything public?